src/Cache/ResultCacheStatement.php

```diff
@@ -240,63 +129,45 @@ public function fetchOne()
      */
     public function fetchAllNumeric() : array
     {
-        if ($this->statement instanceof ForwardCompatibleResultStatement) {
-            $data = $this->statement->fetchAllAssociative();
-        } else {
-            $data = $this->statement->fetchAll(FetchMode::ASSOCIATIVE);
-        }
+        $this->store(
+            $this->statement->fetchAllAssociative()
+        );
 
-        return $this->store($data);
+        return array_map('array_values', $this->data);
```

Comment:
This will need to be backported to `2.11.x` (found by PHPStan and enforced by the refactored `ResultCacheTest`).